dali/dfuplus/dfuplus.cpp

```diff
@@ -178,17 +175,49 @@ bool CDfuPlusHelper::runLocalDaFileSvr(SocketEndpoint &listenep,bool requireauth
 
 bool CDfuPlusHelper::checkLocalDaFileSvr(const char *eps,SocketEndpoint &epout)
 {
+    unsigned port = securitySettings.queryDaliServixPort();
+    unsigned sslport = securitySettings.queryDaliServixSSLPort();
+
+    DAFSConnectCfg connectMethod = securitySettings.queryDAFSConnectCfg();
+
+    unsigned dafsPort;
+    StringBuffer addlPort;
+    if (connectMethod == SSLNone)
+    {
+        dafsPort = port;
+        addlPort.appendf("%u", port);
+    }
+    else if (connectMethod == SSLOnly)
+    {
+        dafsPort = sslport;
+        addlPort.appendf("%u", sslport);
+    }
+    else if (connectMethod == SSLFirst)
+    {
+        dafsPort = sslport;
+        addlPort.appendf("%u:%u", sslport, port);
+    }
+    else
+    {
+        dafsPort = port;
+        addlPort.appendf("%u:%u", port, sslport);
+    }
     if (!eps||!*eps)
-        epout.setLocalHost(securitySettings.queryDaliServixPort());
+        epout.setLocalHost(dafsPort);
     else {
-        epout.set(eps,securitySettings.queryDaliServixPort());
+        epout.set(eps,dafsPort);
         if (!epout.isLocal())
             return false;
     }
-    progress("Checking for local Dali File Server\n");
-    if (!testDaliServixPresent(epout)) // only lookup local
-        runLocalDaFileSvr(epout,false,0);
-    return true;
+    progress("Checking for local Dali File Server on port %s\n", addlPort.str());
+
+    unsigned short nPort = epout.port;
```

Comment:
confusing, because afaics, this nPort value (and epout.port) will never be used.
testDaliServixPresent() will overwrite nPort if successful, and the code below will not use if not successful.